app/src/main/res/layout/fragment_thread.xml

```diff
@@ -22,107 +22,122 @@
     android:layout_height="match_parent"
     tools:context=".ui.main.thread.ThreadFragment">
 
-    <com.google.android.material.appbar.MaterialToolbar
-        android:id="@+id/toolbar"
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Here you have a `ConstraintLayout` with two `ConstraintLayout`s inside. `ConstraintLayout`s are meant to avoid views intrication so here you might be able to just get rid of the two inner `ConstraintLayout`s ?